UGENE-7375. Check user forward and reverse primers.

evelinabis opened the pull request:
Examples of reports:
![0](https://user-images.githubusercontent.com/39720063/130772080-ac00be1e-4ca2-4236-9236-325244301b9e.jpg)
![1](https://user-images.githubusercontent.com/39720063/130772129-d8cf22d6-325a-4db8-9073-e26a3e918d0a.jpg)
![2](https://user-images.githubusercontent.com/39720063/130772139-9da4b89b-08d3-401e-a26a-d744671a0fab.jpg)
![3](https://user-images.githubusercontent.com/39720063/130772150-ef5553e9-25e9-4852-87ae-5a341c7a6e0b.jpg)
![4](https://user-images.githubusercontent.com/39720063/130772168-ac93e791-e376-467e-972d-e1556d7607b3.jpg)
![5](https://user-images.githubusercontent.com/39720063/130772178-7b420b27-cd51-4777-b804-07bc16195807.jpg)
![6](https://user-images.githubusercontent.com/39720063/130772209-830defcd-22f7-43c4-8d9b-51ec48c788e6.jpg)
![7](https://user-images.githubusercontent.com/39720063/130772223-f74bd21b-bd8e-48ec-8f9d-a25f7a36934b.jpg)

Commit:
UGENE-7375. Check user forward and reverse primers.
Added user primers to report.
Changes by review.
Splited "Unwanted connections of user primers" table into two.

Commit:
UGENE-7375. Check user forward and reverse primers.
Refactored.

DmitriiSukhomlinov commented on src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp:
```diff
@@ -43,6 +45,244 @@
 namespace U2 {
 
+//To generate a PCRPrimerDesignForDNAAssemblyTask report.
+namespace ReportUtils {
```
Look, you have a new namespace here, but it's used just in one place by `using namespace ReportUtils;`. It looks a bit over-designed. I suggest you to get rid of this namespace

DmitriiSukhomlinov commented on src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp:
```diff
@@ -43,6 +45,244 @@
+static QString reverseUserPrimerStr();
+
+//No validation of columns number when adding rows.
+class UserPrimersTable {
```
Please, move this class to the separate .h and .cpp files, cuz this one is overfilled already

DmitriiSukhomlinov commented on src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.h:
```diff
@@ -35,6 +35,14 @@ class FindUnwantedIslandsTask;
     Q_OBJECT
 public:
+    //Stores reports of unwanted connections for one user primer.
+    struct UserPrimerReports {
```
It could be moved to the private part

DmitriiSukhomlinov reviewed (changes requested):
Looks good, minor comments

Commit:
UGENE-7375. Check user forward and reverse primers.
Namespace ReportUtils moved to struct PCRPrimerDesignTaskReportUtils in
separate files.

DmitriiSukhomlinov commented on src/plugins/pcr_primer_design_for_dna_assembly/src/utils/PCRPrimerDesignTaskReportUtils.h:
```diff
@@ -0,0 +1,101 @@
+        QString getTable() const;
+    };
+
+    static QString errMsg();
```
The all functions have to have descriptions

DmitriiSukhomlinov commented on src/plugins/pcr_primer_design_for_dna_assembly/src/tasks/PCRPrimerDesignForDNAAssemblyTask.cpp:
```diff
@@ -189,6 +191,7 @@ void PCRPrimerDesignForDNAAssemblyTask::run() {
             break;
         }
     }
+    saveUserPrimersReports();
```
I think, it's better to use a "generate" word. Like `generateUserPrimersReport`

DmitriiSukhomlinov reviewed (approved):
Nice, fix function comments and merge

Commit:
UGENE-7375. Check user forward and reverse primers.
Changes by review and other changes.

evelinabis committed:
Update PCRPrimerDesignTaskReportUtils.h
